.gitignore

```diff
@@ -1,88 +1,77 @@
+# ignore file with wildcard anywhere
 *.swp
 *.pyc
-be/output
-be/build
-be/build_Release
-be/ut_build
-output
-docs/contents
-docs/.temp
-docs/.vuepress/dist
-docs/node_modules
-docs/build
-gensrc/build
-fe/fe-core/target
-thirdparty/src
-thirdparty/installed
-*.so.tmp
-.DS_Store
+*.class
 *.iml
+*.swp
+*.jar
+*.zip
+*.gz
+*.log
+*.so.tmp
+*.flattened-pom.xml
 core.*
-extension/**/.classpath
-extension/**/target
-nohup.out
 
-fe/log
+# ignore file with specifiy name anywhere
+.DS_Store
+.classpath
+nohup.out
 custom_env.sh
-ut_dir
-log/
-fe_plugins/**/target
-fe_plugins/output
-fe/mocked
-fe/**/target
+derby.log
 dependency-reduced-pom.xml
-fe_plugins/**/.classpath
+package-lock.json
+yarn.lock
+
+# ignore all sub `/bin` and `/build` directory
+**/build/
+**/bin/
+**/target/
+**/node_modules/
+**/dist/
+**/log/
+
+# ignore project file
+.cproject
+.project
+.settings/
+**/.idea/
+**/.vscode/
+
+# docs
+docs/contents/
+docs/.temp
+
+# output, thirdparty, extension
+output/
+conf/
+rpc_data/
+thirdparty/src
+thirdparty/installed
+
+fe_plugins/output
 fe_plugins/**/.factorypath
-samples/**/.classpath
-fe/fe-core/src/main/resources/static/
+
 fs_brokers/apache_hdfs_broker/src/main/resources/
 fs_brokers/apache_hdfs_broker/src/main/thrift/
+
+# regression-test
+regression-test/framework/dependency-reduced-pom.xml
+
 samples/doris-demo/remote-udf-python-demo/*_pb2.py
 samples/doris-demo/remote-udf-python-demo/*_pb2_grpc.py
 samples/doris-demo/remote-udf-cpp-demo/*.o
 samples/doris-demo/remote-udf-cpp-demo/*.pb.h
 samples/doris-demo/remote-udf-cpp-demo/*.pb.cc
 samples/doris-demo/remote-udf-cpp-demo/function_server_demo
-regression-test/framework/target
-regression-test/framework/dependency-reduced-pom.xml
 
-# ignore eclipse project file & idea project file
-.cproject
-.project
-.settings/
-.idea/
-/Default/
-be/cmake-build
-be/cmake-build-debug
-be/cmake-build-release
-be/.vscode
-be/src/gen_cpp/*.cc
-be/src/gen_cpp/*.cpp
-be/src/gen_cpp/*.h
+# FE
+fe/mocked/
+fe/fe-core/src/main/resources/static/
+
+# BE
+be/*build*/
```

Comment:
his modification is too broad. Why don't we sort out the output directory and put them all in one directory?